### ISSUE TITLE
Can't get movie details to load

### ISSUE THREAD

**Issue by KitzyKitt** [neo4j-examples/movies-java-spring-data-neo4j#13]
The list of movies is populating, as is the graph.  However, the details for the individual movies isn't loading.

Going to http://localhost:8080/movies/search/findByTitle?title=The%20Matrix%20Reloaded is returning malformed JSON. Nothing seems to have an id.

{
  "title" : "The Matrix Reloaded",
  "released" : 2003,
  "tagline" : "Free your mind",
  "roles" : [ {
    "roles" : [ "Morpheus" ],
    "person" : {
      "name" : "Laurence Fishburne",
      "born" : 1961
    },
    "movie" : { : 9
    }
  }, {
    "roles" : [ "Agent Smith" ],
    "person" : {
      "name" : "Hugo Weaving",
      "born" : 1960
    },
    "movie" : { : 9
    }
  }, {
    "roles" : [ "Trinity" ],
    "person" : {
      "name" : "Carrie-Anne Moss",
      "born" : 1967
    },
    "movie" : { : 9
    }
  }, {
    "roles" : [ "Neo" ],
    "person" : {
      "name" : "Keanu Reeves",
      "born" : 1964
    },
    "movie" : { : 9
    }
  } ],
  "_links" : {
    "self" : {
      "href" : "http://localhost:8080/movies/9"
    },
    "movie" : {
      "href" : "http://localhost:8080/movies/9"
    }
  }
}

**Comment by magemello** [comment]
I have the same problem, any idea?

**Comment by KitzyKitt** [comment]
I did figure out a way to get it to work.  I posted the solution at http://stackoverflow.com/questions/42676594/neo4j-movies-example-returning-malformed-json

**Comment by jexp** [comment]
That's really weird, this is supposed to be part of the SD REST integration. 
This used to work for SDN 3.x we have to look into this.

**Comment by magemello** [comment]
Hi @jexp There is any issue that I can follow on the main repo...to check when this issue will be fixed?

**Comment by mangrish** [comment]
@magemello @jexp @KitzyKitt: @nmervaillie and myself will take time to look into this today. We will update this ticket once we know what the issues are (and raise a ticket to follow in either OGM GH or SDN JIRA).

**Comment by magemello** [comment]
If can help...I had this error also in another app build by me, and the problem is always related to a @Relationship entity...I guess something wrong in the deserializer there...

**Comment by magemello** [comment]
Hi @mangrish do you have any news?

**Comment by mangrish** [comment]
@magemello: Still looking into it. I hope to have something ready by Monday.

**Comment by magemello** [comment]
Thank you @mangrish for your update, and sorry for pressuring you but a similar issue is blocking me on another project

**Comment by magemello** [comment]
I have tried to manually return the findAll in the @RestController and the nested object are returned only the first time, the second time instead of the full object only the id is returned....this is  related to how jackson does the deserialization...maybe the same problem is in the @RepositoryRestResource?

**Comment by magemello** [comment]
removing from all the domain classes:
```
@JsonIdentityInfo(generator=ObjectIdGenerators.PropertyGenerator.class, property="id")
```

and adding @JsonIgnore on @EndNode Movie of the Role class ti works.

I don't think it's a solution...but it's a workaround

**Comment by mangrish** [comment]
All the issues are caused by this bug: https://github.com/FasterXML/jackson-databind/issues/1298.

We can't fix this until this is fixed really.

I also tried working around this with JSOG but it too is affected by the same bug.

**Comment by mangrish** [comment]
This was actually caused by an error in SDN. See https://github.com/spring-projects/spring-data-neo4j/commit/1631ef4f18d2b8f974a2381c6f8f0a1ebe4f03a3. This has been fixed in 4.2.2-SNAPSHOT and 3.0.0-SNAPSHOT
